src/platform/backends/lxd/lxd_vm_image_vault.cpp

```diff
@@ -136,29 +212,31 @@ mp::VMImage mp::LXDVMImageVault::fetch_image(const FetchType& fetch_type, const
     }
     catch (const LXDNotFoundException&)
     {
-        QJsonObject source_object;
-
-        source_object.insert("type", "image");
-        source_object.insert("mode", "pull");
-        source_object.insert("server", info.stream_location);
-        source_object.insert("protocol", "simplestreams");
-        source_object.insert("image_type", "virtual-machine");
-
-        if (id.startsWith(QString::fromStdString(query.release)))
+        auto lxd_image_hash = get_lxd_image_hash_for(id);
+        if (!lxd_image_hash.empty())
         {
-            source_object.insert("fingerprint", id);
+            source_image.id = lxd_image_hash;
         }
-        else
+        else if (!info.stream_location.isEmpty())
         {
-            source_object.insert("alias", QString::fromStdString(query.release));
+            lxd_download_image(id, info.stream_location, QString::fromStdString(query.release), monitor);
         }
+        else
```

Comment:
Feels weird as a fall-through, should we be asserting something here?

Reply:
Hmm, I'm not sure what we should be asserting.  This is the last way to try to get an image and we know that it's a legitimate image since we got it from a manifest.  Otherwise, we would have already failed before getting here.

Reply:
Sure, if we assume that we're using the vault correctly, nothing below will fail, but maybe due dilligence to assert `info.image_location` isn't empty? Your call.

Reply:
Ok, I changed this to account for `info.image_location` and `throw` if none of the conditions can be met.